R/notebooks/gaussian_mix_uni.Rmd

```diff
@@ -0,0 +1,237 @@
+---
+title: "BayesMixR - Univariate Gaussian Mixture"
+output: html_notebook
+---
+
+```{r}
+library("bayesmixr")
+build_bayesmix()
+```
+
+## Generate Data
+
+```{r}
+data <- c(rnorm(100, -3, 1),
+          rnorm(100, 3, 1))
+hist(data, probability = TRUE, main = "", xlab = "")
+```
+
+## The Bayesian Model
+
+$$
+\begin{align*}
+y_i \mid \theta_i=(\mu_i, \sigma^2_i) & \sim \mathcal{N}(\mu_i, \sigma^2_i) \\
+\theta_i \mid P & \sim P \\
+P & \sim DP(\alpha G_0)
+\end{align*}
+$$
+
+and $G_0(d\mu, d\sigma^2) = \mathcal N(d\mu | \mu_0, \sigma^2/\lambda) \times \mathcal{IG}(\sigma^2 | a, b)$
+
+We consider different prior specifications
+
+### Fixed hyperparameters
+
+$\alpha = 1$ and $(\mu_0,~\lambda,~a,~b) = (0,~0.1,~2,~2)$
+
+```{r}
+dp_params_fix =
+"
+fixed_value {
+    totalmass: 1.0
+}
+"
+
+g0_params_fix =
+"
+fixed_values {
+    mean: 0.0
+    var_scaling: 0.1
+    shape: 2.0
+    scale: 2.0
+}
+"
+```
+
+### Prior on $\alpha$ and $\mu_{0}$
+
+$$
+\begin{align*}
+\alpha &\sim \text{Gamma}(2, 2) \\
+\mu_0 &\sim \mathcal{N}(0, 10) \\
+(\lambda,~a,~b) &= (0.1,~2,~2)
+\end{align*}
+$$
+
+```{r}
+dp_params_prior =
+"
+gamma_prior {
+  totalmass_prior {
+    shape: 4.0
+    rate: 2.0
+  }
+}
+"
+
+g0_params_meanprior =
+"
+normal_mean_prior {
+    mean_prior {
+        mean: 0.0
+        var: 10.0
+    }
+    var_scaling: 0.1
+    shape: 2.0
+    scale: 2.0
+}
+"
+
+dp_params = c(dp_params_fix, dp_params_prior)
+dp_names = c("fix", "gamma prior")
+```
+
+### **Prior on all the hyperparameters**
+
+$$
+\begin{align*}
+\alpha &\sim \text{Gamma}(2,~2) \\
+\mu_0 &\sim \mathcal{N}(0,~10) \\
+\lambda &\sim \text{Gamma}(0.2,~0.6) \\
+a &= 1.5 \\
+b &\sim \text{Gamma}(4,~2)
+\end{align*}
+$$
+
+```{r}
+g0_params_allprior =
+"
+ngg_prior {
+    mean_prior {
+        mean: 5.5
+        var: 2.25
+    }
+    var_scaling_prior {
+        shape: 0.2
+        rate: 0.6
+    }
+    shape: 1.5
+    scale_prior {
+        shape: 4.0
+        rate: 2.0
+    }
+}
+"
+
+g0_params = c(g0_params_fix, g0_params_meanprior, g0_params_allprior)
+g0_names = c("Fix", "Mean prior", "NGG prior")
+```
+
+## The algorithm
+
+We consider all available algorithms in bayesmix: Neal's Algorithms 2, 3 and 8
```

Comment:
Poor Split&Merge and BlockedGibbs :(

Reply:
True, @brunoguindani. We should quickly fix the text in the notebook or add run with S&M and BG. What do you prefer?

Reply:
It's not necessary to use all algorithms since it is just an example, but we at least have to mention that this are not "all" the algorithms in the library

Reply:
Totally agree, I'll go with that.